src/constants.ts

```diff
@@ -138,7 +149,9 @@ export const action: ActionInterface = {
     ? true
     : getInput('ssh-key'),
   targetFolder: getInput('target-folder'),
-  workspace: process.env.GITHUB_WORKSPACE || ''
+  workspace: process.env.GITHUB_WORKSPACE || '',
+  add_tag: !isNullOrUndefined(getInput('tag')),
```

Comment:
Is `add_tag` and `externalRepositoryTarget` necessary? Are we not able to check the existence of `tag` and `repositoryName` instead directly?

Reply:
Hi @JamesIves , I do not know how to do that. Ideally yes, only checking the existence of those should be sufficient.

I'm happy to implement this change if some guidance is provided.

Reply:
I think we can just check to see if these fields have content or not, we do this in a few places by just validating the existence of content; see: https://github.com/JamesIves/github-pages-deploy-action/blob/dev/src/git.ts#L152 for example